core/src/main/java/com/linecorp/armeria/internal/server/ResponseConversionUtil.java

```diff
@@ -86,6 +86,10 @@ private static HttpResponseWriter aggregateFrom(CompletableFuture<?> future,
                                                     ResponseHeaders headers, HttpHeaders trailers,
                                                     Function<Object, HttpData> contentConverter) {
         final HttpResponseWriter writer = HttpResponse.streaming();
+        writer.whenComplete().handle((ignored, cause) -> {
+            future.completeExceptionally(cause);
```

Comment:
How about calling this only when `cause` is not null?

Reply:
My mistake, 😅 I will fix it. Thanks for catching this.